unittests/AST/ASTImporterTest.cpp

```diff
@@ -97,6 +97,11 @@ testImport(const std::string &FromCode, const ArgVector &FromArgs,
   ASTContext &FromCtx = FromAST->getASTContext(),
       &ToCtx = ToAST->getASTContext();
 
+  FromAST->getDiagnostics().getClient()->BeginSourceFile(
+      FromCtx.getLangOpts(),FromAST->getPreprocessorPtr().get());
```

Comment:
Shouldn't we add this code to mare places? E.g.: to the fixtures.  The formatting is a bit off, use spaces after the commas in the argument list.

Reply:
The code can be added to other places in getImportedDecl and Import. But probably these calls should belong into ASTUnit somewhere to a hard-to-find-out place (`ASTUnit::~ASTUnit` calls EndSourceFile).

Reply:
I do not know if it is better to put the BeginSourceFile call into `ASTBuilderAction::runInvocation` or into `buildASTFromCodeWithArgs`. (But no EndSourceFile is called then.)